Spring Cross-Site Request Forgery (CSRF)

bhaskarvilles opened the pull request:
The ```body``` parameter is vulnerable to Cross Site Request Forgery (CSRF) attacks due to not using _Spring Security_. This could allow an attacker to execute requests on a user's behalf. Consider including Spring Security's CSRF protection within your application.
and
Some version bumps.

snyk-bot committed:
fix: upgrade io.sentry:sentry-logback from 5.6.3 to 5.7.0

Snyk has created this PR to upgrade io.sentry:sentry-logback from 5.6.3 to 5.7.0.

See this package in Maven Repository:
https://mvnrepository.com/artifact/io.sentry/sentry-logback/

See this project in Snyk:
https://app.snyk.io/org/bhaskarvilles/project/603f6d6e-1273-4b00-afd6-fe7fce71a890?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade io.sentry:sentry-spring-boot-starter from 5.6.3 to 5.7.0

Snyk has created this PR to upgrade io.sentry:sentry-spring-boot-starter from 5.6.3 to 5.7.0.

See this package in Maven Repository:
https://mvnrepository.com/artifact/io.sentry/sentry-spring-boot-starter/

See this project in Snyk:
https://app.snyk.io/org/bhaskarvilles/project/603f6d6e-1273-4b00-afd6-fe7fce71a890?utm_source=github&utm_medium=referral&page=upgrade-pr

bhaskarvilles committed:
Merge pull request #4 from bhaskarvilles/snyk-upgrade-57274178e552ca041bae536056632198

[Snyk] Upgrade io.sentry:sentry-spring-boot-starter from 5.6.3 to 5.7.0

bhaskarvilles committed:
Merge pull request #3 from bhaskarvilles/snyk-upgrade-f08c65daf7dd00c48f01445344236eb9

[Snyk] Upgrade io.sentry:sentry-logback from 5.6.3 to 5.7.0

Commit:
Spring Cross-Site Request Forgery (CSRF)

snyk-bot committed:
fix: upgrade io.sentry:sentry-spring-boot-starter from 5.7.0 to 5.7.3

Snyk has created this PR to upgrade io.sentry:sentry-spring-boot-starter from 5.7.0 to 5.7.3.

See this package in Maven Repository:
https://mvnrepository.com/artifact/io.sentry/sentry-spring-boot-starter/

See this project in Snyk:
https://app.snyk.io/org/bhaskarvilles/project/603f6d6e-1273-4b00-afd6-fe7fce71a890?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade io.sentry:sentry-logback from 5.7.0 to 5.7.2

Snyk has created this PR to upgrade io.sentry:sentry-logback from 5.7.0 to 5.7.2.

See this package in Maven Repository:
https://mvnrepository.com/artifact/io.sentry/sentry-logback/

See this project in Snyk:
https://app.snyk.io/org/bhaskarvilles/project/603f6d6e-1273-4b00-afd6-fe7fce71a890?utm_source=github&utm_medium=referral&page=upgrade-pr

mike-audi commented:
Will test to confirm fix. 

It should be noted that this API is not publicly available. Only TIKI's ingestion service can post to the KGraph. 

However we do have other endpoints that are likely subject to the same vulnerability in the bouncer & ingestion service.

bhaskarvilles commented:
> Will test to confirm fix. 
> 
> It should be noted that this API is not publicly available. Only TIKI's ingestion service can post to the KGraph. 
> 
> However we do have other endpoints that are likely subject to the same vulnerability in the bouncer & ingestion service. 

Okay Mike also please check the version bumps.

bhaskarvilles committed:
Merge pull request #5 from bhaskarvilles/snyk-upgrade-203fd00ea42093cccd859ef5e25d7f6f

[Snyk] Upgrade io.sentry:sentry-spring-boot-starter from 5.7.0 to 5.7.3

bhaskarvilles committed:
Merge pull request #6 from bhaskarvilles/snyk-upgrade-ba93efad5de25a22cead3e633bdf757a

[Snyk] Upgrade io.sentry:sentry-logback from 5.7.0 to 5.7.2

snyk-bot committed:
fix: upgrade io.sentry:sentry-logback from 5.7.2 to 5.7.3

Snyk has created this PR to upgrade io.sentry:sentry-logback from 5.7.2 to 5.7.3.

See this package in Maven Repository:
https://mvnrepository.com/artifact/io.sentry/sentry-logback/

See this project in Snyk:
https://app.snyk.io/org/bhaskarvilles/project/603f6d6e-1273-4b00-afd6-fe7fce71a890?utm_source=github&utm_medium=referral&page=upgrade-pr

bhaskarvilles commented:
> Will test to confirm fix. 
> 
> It should be noted that this API is not publicly available. Only TIKI's ingestion service can post to the KGraph. 
> 
> However we do have other endpoints that are likely subject to the same vulnerability in the bouncer & ingestion service. 


Testing ?

bhaskarvilles commented:
Any update on this @mike-audi  ?

mike-audi commented:
Not yet, it's not a pressing issue (see my earlier comment). Will look to close asap anyways. 

The endpoint is not public. It's only reached via another service of ours so while good practice to put the change in, it's not at risk.

bhaskarvilles committed:
Merge pull request #7 from bhaskarvilles/snyk-upgrade-86a6297c29fe4d232afc00ac0811447c

Upgrade io.sentry:sentry-logback from 5.7.2 to 5.7.3
